syncapi/storage/shared/storage_sync.go

```diff
@@ -264,6 +267,21 @@ func (d *DatabaseTransaction) GetEventsInTopologicalRange(
 
 	// Retrieve the events' contents using their IDs.
 	events, err = d.OutputEvents.SelectEvents(ctx, d.txn, eIDs, filter, true)
+	if err != nil {
+		return
+	}
+	// Check if we should be able to return events.
+	// If we received 0 events, this most likely means that the provided filter removed them.
+	if len(eIDs) > 0 && len(events) == 0 {
+		// We try to fetch the events without a filter, so we can tell the client if there
+		// are more events earlier than the requested and filtered.
+		events, err = d.OutputEvents.SelectEvents(ctx, d.txn, eIDs, nil, true)
```

Comment:
This doesn't make sense. If we are filtering out events, why are we returning some events?